File: components/pages/radar.js
import { html } from 'https://cdn.jsdelivr.net/gh/lit/dist@2/core/lit-core.min.js';
import { QuestionElement } from '../base/question-element.js';
import { RADAR_MODE } from '../../game-pages.js';

class RadarMode extends QuestionElement {
  render() {
    const title = RADAR_MODE.categories[0].title;
    const inputs = RADAR_MODE.categories[0].fields.map(
      (item) =>
        html`<persistent-input id="${RADAR_MODE.title}.${item.id}" gameId="${this.gameId}"
          >${item.label}</persistent-input
        >`,
    );

    return html`
      <div class="game-page">
        <div class="game-header">
          <div class="game-title">${RADAR_MODE.title}</div>
          <div class="game-info">
            Cost: <span>Draw 2, Pick 1</span> Time: <span>5 Minutes</span>
          </div>
        </div>

        <div class="game-question">Are you within <span class="light">Distance</span> of me?</div>
        <div class="category">
          <div class="category-title">${title}</div>
          ${inputs}
          <persistent-input-custom-label
            id="radar.choose"
            gameId="${this.gameId}"
            label="🎯CHOOSE"
          />
        </div>
      </div>
    `;
  }
}

customElements.define('radar-mode', RadarMode);

File: src/pages/radar.rs
use crate::{
    components::{PersistentInput, PersistentInputCustomLabel},
    constants,
};
use yew::prelude::*;

#[function_component(Radar)]
pub fn radar_mode() -> Html {
    let constants::GameMode { title, categories } = constants::radar_mode();

    let inputs = categories[0]
        .fields
        .iter()
        .filter(|item| item.id != "choose")
        .map(|item|
            html! { <PersistentInput id={format!("{}.{}", title, item.id)} label={item.label} /> }
        );

    html! {
        <div class="game-page">
            <div class="game-header">
                <div class="game-title">{ title }</div>
                <div class="game-info">
                    { "Cost: " }
                    <span>{ "Draw 2, Pick 1" }</span>
                    { " Time: " }
                    <span>{ "5 Minutes" }</span>
                </div>
            </div>
            <div class="game-question">
                { "Are you within " }
                <span class="light">{ "Distance" }</span>
                { " of me?" }
            </div>
            <div class="category">
                <div class="category-title">{ categories[0].title }</div>
                { for inputs }
                <PersistentInputCustomLabel id="radar.choose" label="🎯CHOOSE" />
            </div>
        </div>
    }
}
